fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/AndOrComponent.java

```diff
@@ -83,4 +85,39 @@ private Boolean evalInternal(@Nonnull Function<QueryComponent, Boolean> evalChil
     public boolean isAsync() {
         return getChildren().stream().anyMatch(QueryComponent::isAsync);
     }
+
+    @Nullable
+    @Override
+    @API(API.Status.EXPERIMENTAL)
+    public ExpressionRef<QueryComponent> asNestedWith(@Nonnull NestedContext nestedContext,
+                                                      @Nonnull ExpressionRef<QueryComponent> thisRef) {
+        // An AndComponent can be placed in a NestedContext if and only if all of its children are nested under the
+        // context's parent.
+        if (nestedContext.isParentFieldRepeated()) {
```

Comment:
I'm not sure that "repeated" here is exactly right (here and elsewhere in this PR). It's sort of "repeated with `FanType.FanOut`" I think implicitly. Truth be told, I haven't thought too much about how `FanType.Concatenate` interacts with any of this. We don't have an `AllOfThem` component, nor do we have the ability to execute `OneOfThem`s by evaluating a predicate on the concatenated entries of the index, but I guess I just want to make sure that they will fit in in some reasonable way (if that makes sense).